test/xpu/nn/test_packed_sequence_xpu.py

```diff
@@ -15,12 +15,15 @@
 
     def myxpu(self, *args, **kwargs):
 
-        ex = torch.tensor((), dtype=self.data.dtype, device=self.data.device).to(*args, **kwargs)
-        if ex.device.type == 'xpu':
+        ex = torch.tensor((), dtype=self.data.dtype, device=self.data.device).to(
+            *args, **kwargs
+        )
+        if ex.is_xpu:
             return self.to(*args, **kwargs)
-        return self.to(*args, device='xpu', **kwargs)
+        kwargs["device"] = "xpu"
+        return self.to(*args, **kwargs)
 
-    rnn_utils.PackedSequence.xpu = types.MethodType(rnn_utils.PackedSequence, myxpu)
+    rnn_utils.PackedSequence.xpu = myxpu
```

Comment:
What's the difference? Looks logics are same?

Reply:
types.MethodType is for object, but we want to change a class function member. so we use rnn_utils.PackedSequence.xpu = myxpu.